### ISSUE TITLE
Can't shoot while moving full speed with mouse pointer at top of screen.

### ISSUE THREAD

**Issue by jbum** [BZFlag-Dev/bzflag#30]
I'm using 2.4.4 RC2, for Mac OSX.   After emptying my clip, when my bullets reload, there is a delay of a 2-3 seconds before I can fire.  This delay occurs after the bullets have visually reloaded.

There are times when I can't fire, and I haven't recently fired, but I can reproduce it most easily by firing.

EDIT: The bug seems more correlated to my speed than to reload or spawning.  If I am driving forward at full speed, I often can't fire, but then when I stop, I can fire again.

**Comment by allejo** [comment]
I haven't been able to reproduce this. Were you on a specific map where this occurred? What version of OS X are you using? Also, would you mind sharing your configuration file (be sure to remove your password from it)?

**Comment by jbum** [comment]
Happens on all the maps I try.  Also,often can't fire the first time I try to shoot after spawning.  Then, a second or two later, it's fine.

I'm on OSX 10.10.5.  

Heres my config:  http://pastebin.com/jdUrSdCc

**Comment by macsforme** [comment]
I have your same operating system, and I tried your configuration, but could not reproduce the issue.

Are you by any chance using a non-U.S. keyboard? Or do you use any custom drivers or tools that affect input?

Are you able to try a clean install of OS X (perhaps on a spare disk), and see if you still experience the issue?

**Comment by jbum** [comment]
I'm using a US keyboard.  I'll try it on a different mac.  If it's a custom driver, it only has this interaction with this version of Bzflag.  2.4.2 doesn't exhibit the problem.  

I just noticed that the problem usually crops up when I am moving at full speed towards an enemy.  I am playing on maps like clay hills & paradise that apply tank velocity to bullets, so I'm wondering if there is some code that prevents firing when it thinks the bullets will go too fast?  When I slow down, I can shoot again.

**Comment by macsforme** [comment]
Confirmed that moving the cursor to the extreme top position of the screen (in fullscreen mode) causes all mouse button clicks to not register, and also moving the cursor to the extreme top, bottom, left or right position of the window (in windowed mode) causes all mouse button clicks to not register.

**Comment by macsforme** [comment]
I isolated the issue, and it is in SDL code. Since input events are being discarded before they even reach BZFlag's code, it does not appear to be possible to create a workaround for it in our code. I reported the bug to the SDL team with a patch to correct it (https://bugzilla.libsdl.org/show_bug.cgi?id=3190), and if it isn't corrected soon, we can always patch the SDL framework that we ship with BZFlag 2.4.4.
